### PR TITLE
Support custom units for custom measurements

### LINKED ISSUE #1053: Support custom units for custom measurements
### Description

Due to the limitations of `enums` on Dart, there was no easy way to extend the enum and accept a plain String.
We can leverage this feature if needed, when we upgrade to Dart 2.17, or just set the `unit` as a plain string and document all the possible values.

Related to https://github.com/getsentry/sentry-dart/pull/1011

Example on Java:

```
transaction.setMeasurement("metric1", 2, MeasurementUnit.Duration.DAY)
transaction.setMeasurement("metric1", 2, MeasurementUnit.Custom("myOwnUnit"))
```

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Closes #1053

## :green_heart: How did you test it?

- Unit Tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by denrase** [comment]
@marandaneto Do we also have to lowercase the custom measurements provided by constructor? Java does this: https://github.com/getsentry/sentry-java/blob/54cebc81e18ff2f50da062cac751681e7e4e83d4/sentry/src/test/java/io/sentry/MeasurementUnitTest.kt#L46

**Comment by marandaneto** [comment]
> @marandaneto Do we also have to lowercase the custom measurements provided by constructor? Java does this: [getsentry/sentry-java@`54cebc8`/sentry/src/test/java/io/sentry/MeasurementUnitTest.kt#L46](https://github.com/getsentry/sentry-java/blob/54cebc81e18ff2f50da062cac751681e7e4e83d4/sentry/src/test/java/io/sentry/MeasurementUnitTest.kt#L46)

Looking at https://develop.sentry.dev/sdk/event-payloads/transaction/ I don't see anything in the spec.
Maybe @romtsn can tell us if its written somewhere or the reasons?

**Comment by romtsn** [comment]
> Maybe @romtsn can tell us if its written somewhere or the reasons?

No specific reason, we just did not special-case it and used the same method `apiName` which lowercases the value. But looking at relay, they also [lowercase](https://github.com/getsentry/relay/blob/ff2d211450ebb89e23c9a2855f87213bd27be283/relay-common/src/constants.rs#L530) it anyway, so I guess it doesn't matter what we send

File: dart/lib/src/sentry_measurement_unit.dart [hunk 1]
```diff
@@ -1,152 +1,124 @@
...
 
   /// Day (`"day"`), 86,400 seconds.
-  day,
+  day(name: 'day'),
 
   /// Week (`"week"`), 604,800 seconds.
-  week,
+  week(name: 'week');
+
+  const DurationSentryMeasurementUnit({required this.name});
```
Comment (marandaneto) [inline dart/lib/src/sentry_measurement_unit.dart]: Instead of using `required`, we can just use a single non named param.

**Review by marandaneto — APPROVED** [review]
Left a comment https://github.com/getsentry/sentry-dart/pull/1181/files#r1047071003 otherwise LGTM

**Pull request merged**